src/main/webapp/script.js

```diff
@@ -20,28 +20,22 @@ function listElections() {
 
   fetch('/election')
     .then(response => response.json())
-    .then((textResponse) => {
-      let electionList = JSON.parse(textResponse).elections;
+    .then((electionList) => {
       // ocdDivisionId is in the form "ocd-division/country:us/state:<state_code>"
       // For example, Lousiana's ID is "ocd-division/country:us/state:la"
       let regex = /ocd\-division\/country\:us\/(state|district)\:([a-z][a-z])/;
 
       // For every election returned, store it if the state matches or if national election,
       // to later display the details to the user.
       electionList.forEach((election) => {
-        let ocdId = election.ocdDivisionId;
+        let ocdId = election.scope;
 
         // Match the OCD Division ID with the regular expressions to check if this election
         // is a state election, district election, or national election.
-        let stateMatch = regex.exec(ocdId);
-        let districtMatch = regex.exec(ocdId);
+        let regexMatch = regex.exec(ocdId);
 
-        if (stateMatch != null) {
-          if (stateMatch[1] == selectedStateId) {
-            stateElections.push(election);
-          }
-        } else if (districtMatch != null) {
-          if (districtMatch[1] == selectedStateId) {
+        if (regexMatch != null) {
+          if (regexMatch[2] == selectedStateId) {
```

Comment:
combine into:

if (regexMatch != null && regexMatch[2] == selectedStateId) {
} else {
}
?

Should generally try to avoid deep nesting if you can, studies have shown that it greatly reduces readability. Readability is important because studies have also shown the vast majority of time spent in mature products developing is actually spent reading code instead of the actual writing of code. (see 'Clean Code' book for more in depth on those ideas).

Reply:
There's an else condition that I wanted to avoid -- if regexMatch[2] != null and regexMatch[2] != selectedStateId.
In that case, we don't want to save it at all, because the state doesn't match. The else condition right now handles if the election is not a state or district election (national election).

There's potentially a different way to handle that, though -- maybe a separate regex specific to national elections so that the else doesn't mistakenly capture a different edge case. I'll make that adjustment!

Reply:
ACK, ah I see.